tests/conversion_integration.sh

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+set -eux
+
+tmp_dir=$(mktemp -d /tmp/run_conversion.XXXXXX)
+TESTDIR="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+
+function convert_and_verify() {
+  ffp-convert "${1}" -f "${2}" "${3}" -t "${4}"
+  python "${TESTDIR}"/conversion_integration.py "${1}" "${2}" "${3}" "${4}"
+}
+
+function verify_all_conversions() {
+  local input=${1}
+  local in_format=${2}
+  local out_path_prefix=${tmp_dir}/${in_format}
+  convert_and_verify "${input}" "${in_format}" "${out_path_prefix}_to_fifu.fifu" finalfusion
+  convert_and_verify "${input}" "${in_format}" "${out_path_prefix}_to_w2v.w2v" word2vec
+  convert_and_verify "${input}" "${in_format}" "${out_path_prefix}_to_ft.bin" fasttext
+  convert_and_verify "${input}" "${in_format}" "${out_path_prefix}_to_text.txt" text
+  convert_and_verify "${input}" "${in_format}" "${out_path_prefix}_to_text.dims.txt" textdims
+}
+
+# txt dims
+input="${TESTDIR}/data/embeddings.dims.txt"
+verify_all_conversions "${input}" textdims
+
+# txt
+input="${TESTDIR}/data/embeddings.txt"
+verify_all_conversions  "${input}" text
+
+# w2v
+input="${TESTDIR}/data/embeddings.w2v"
+verify_all_conversions  "${input}" word2vec
+
+# fifu
+input="${TESTDIR}/data/embeddings.fifu"
+verify_all_conversions  "${input}" finalfusion
+
+# fasttext doesn't support fifu bucket indexers
+# so we're making explicit calls for the other formats
+input="${TESTDIR}/data/ff_buckets.fifu"
+convert_and_verify "${input}" finalfusion \
+  "${tmp_dir}/fifu_buckets_to_fifu.fifu" finalfusion
+convert_and_verify "${input}" finalfusion \
+  "${tmp_dir}/fifu_buckets_to_w2v.w2v" word2vec
+convert_and_verify "${input}" finalfusion \
+  "${tmp_dir}/fifu_buckets_to_text.txt" text
+convert_and_verify "${input}" finalfusion \
+  "${tmp_dir}/fifu_buckets_to_text.dims.txt" textdims
+
+input="${TESTDIR}/data/fasttext.bin"
+verify_all_conversions  "${input}" fasttext
+
+function finish() {
+  rm -rf "$tmp_dir"
+}
+
+trap finish EXIT
```

Comment:
Don't you want to set the trap immediately after creating the temporary directory?

Reply:
I moved it up, but behaviour with `trap` at the end worked as expected on failure:

```
++ mktemp -d /tmp/run_conversion.XXXXXX
+ tmp_dir=/tmp/run_conversion.HIYH5d
+ trap finish EXIT
+++ dirname tests/conversion_integration.sh
++ cd tests
++ pwd -P
+ TESTDIR=/data/python_projects/ffp-finalfusion/tests
+ input=/data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt
+ verify_all_conversions /data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt textdims
+ local input=/data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt
+ local in_format=textdims
+ local out_path_prefix=/tmp/run_conversion.HIYH5d/textdims
+ convert_and_verify /data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt textdims /tmp/run_conversion.HIYH5d/textdims_to_fifu.fifu finalfusion
+ ffp-convert /data/python_projects/ffp-finalfusion/tests/data/embeddings.dims.txt -f textdims /tmp/run_conversion.HIYH5d/textdims_to_fifu.fifu -t finalfusion
tests/conversion_integration.sh: line 15: ffp-convert: command not found
+ finish
+ rm -rf /tmp/run_conversion.HIYH5d
```

Reply:
For me, it doesn't seem to:

```
patching script interpreter paths in tests/conversion_integration.sh
tests/conversion_integration.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/x7fr0bvnwvqvr3zg60di9jxvfwimcw7m-bash-4.4-p23/bin/bash"
++ mktemp -d /tmp/run_conversion.XXXXXX
+ tmp_dir=/tmp/run_conversion.y1K1vI
+++ dirname tests/conversion_integration.sh
++ cd tests
++ pwd -P
+ TESTDIR=/build/source/tests
+ input=/build/source/tests/data/embeddings.dims.txt
+ verify_all_conversions /build/source/tests/data/embeddings.dims.txt textdims
+ local input=/build/source/tests/data/embeddings.dims.txt
+ local in_format=textdims
+ local out_path_prefix=/tmp/run_conversion.y1K1vI/textdims
+ convert_and_verify /build/source/tests/data/embeddings.dims.txt textdims /tmp/run_conversion.y1K1vI/textdims_to_fifu.fifu finalfusion
+ ffp-convert /build/source/tests/data/embeddings.dims.txt -f textdims /tmp/run_conversion.y1K1vI/textdims_to_fifu.fifu -t finalfusion
tests/conversion_integration.sh: line 8: ffp-convert: command not found
builder for '/nix/store/gxjciq20a4mr8nsgbysdd6lzikl543kf-python3.7-ffp-0.0.0.drv' failed with exit code 127
```

Reply:
> I moved it up, but behaviour with `trap` at the end worked as expected on failure:
> 
> ```
> ++ mktemp -d /tmp/run_conversion.XXXXXX
> + tmp_dir=/tmp/run_conversion.HIYH5d
> + trap finish EXIT

Note that you are testing the version here with trap at the top ;).

Reply:
Oops. Well then it should be working properly now!

Reply:
So the `-x` was useful for something after all ;)